src/p5.dance.js

```diff
@@ -247,7 +290,14 @@ module.exports = class DanceParty {
     }
   }
 
-  play(songData, callback) {
+  async play(songData, callback) {
+    if (!this.lastEnsureSpritesPromise_) {
+      // The host has never called ensureSpritesAreLoaded(). We will call it ourselves here,
+      // effectively ensuring that all sprites are loaded.
+      this.ensureSpritesAreLoaded();
+    }
+    await this.lastEnsureSpritesPromise_;
```

Comment:
This also worries me.  We're sort-of guarding against multiple play calls while we are loading sprites, but in that case when the promise resolves don't we run the rest of `play()` multiple times back-to-back?  Seems like instead we should have a simple early-out if a previous play call has not completed.

Reply:
This was intended to deal with the case of new sprites being needed (via a new call to `ensureSpritesAreLoaded()`) (for example, the student’s program changed and added a new character) - we don’t want the play() to start until that is complete.

We can change the contract to require that the caller called (and waited for `ensureSpritesAreLoaded()`). In which case, this isn’t needed and the play() call doesn’t need to return a promise. The CDO repo PR has made that the contract, so it’s just a matter of changing all of the tests in this repo.

Reply:
Updated with the 2nd approach, requiring `ensureSpritesAreLoaded()` to be called.